src/screens/main.screen.tsx

```diff
@@ -37,69 +43,88 @@ const MainScreen = ({ navigation }: ScreenProps) => {
+          <ImageBackground style={styles.bg} source={appBackground}>
             <AppBar location={location} navigation={navigation} />
-            <Text>Loading...</Text>
+            <ScrollView>
+              <GlassView glassStyle={styles.glassWrapper} blurStyle={{ blurAmount: 8, blurType: 'light' }}>
+                <View style={styles.opacity}>
+                  <TouchableOpacity onPress={onSelectToday}>
+                    <Today daySummary={preparedForecast.atDay(today)} />
+                  </TouchableOpacity>
+                  <FiveDays name={location} startDate={today.plus({days:1})} preparedForecast={preparedForecast} onClick={onSelectDay(location)} />
+                </View>
+              </GlassView>
+              {/* { testErr && <ErrorNotification message="Forecast update failed." onClose={() => setTestErr(false)}/>} */}
+            </ScrollView>
           </ImageBackground>
         </View>
       </SafeAreaView>
     );
   }
 
-  const preparedForecast = new Forecast(forecast)
-  const today = DateTime.now()
-  const todaysForecast = preparedForecast.atDay(today)
-
-  const onSelectToday = () =>
-    navigation.navigate(SCREENS.Hourly, {
-      location: location,
-      daySummary: preparedForecast.atDay(today),
-      title: 'Hourly Today'
-    })
-  const onSelectDay = (location: string) =>
-    (day: DateTime, preparedForecast: Forecast) =>
-      navigation.navigate(SCREENS.Hourly, {
-        location: location,
-        daySummary: preparedForecast.atDay(day),
-        title: day.toLocaleString({ weekday: 'long' })
-      });
+  if (forecastError){
+    return (
+      <SafeAreaView>
+        <View style={styles.wrapper}>
+          <ImageBackground source={appBackground} style={styles.bg}>
+            <AppBar location={location} navigation={navigation} />
+            <Text>There was a problem getting the forecast.</Text>
+          </ImageBackground>
+        </View>
+      </SafeAreaView>
+    )
+  }
 
   return (
     <SafeAreaView>
       <View style={styles.wrapper}>
         <ImageBackground source={appBackground} style={styles.bg}>
           <AppBar location={location} navigation={navigation} />
-          <ScrollView>
-            <GlassView glassStyle={styles.glassWrapper} blurStyle={{ blurAmount: 8, blurType: 'light' }}>
-              <View style={styles.opacity}>
-                <TouchableOpacity onPress={onSelectToday}>
-                  <Today daySummary={todaysForecast} />
-                </TouchableOpacity>
-                <FiveDays name={location} startDate={today.plus({ days: 1 })} preparedForecast={preparedForecast} onClick={onSelectDay(location)} />
-              </View>
-            </GlassView>
-          </ScrollView>
         </ImageBackground>
       </View>
     </SafeAreaView>
   );
 }
 
+type ErrorNotificationProps = {
+  message: string,
+  onClose: () => void,
+}
+const ErrorNotification = ({ message, onClose }: ErrorNotificationProps) => {
```

Comment:
I know, also not sure about that. I kept it in because I assumed it would be easier to start on the presentation issue when that code is already there, but not sure.